corehq/apps/translations/forms.py

```diff
@@ -213,58 +213,73 @@ def clean(self):
 
     @classmethod
     def form_for(cls, form_action):
-        if form_action == 'create':
-            return CreateAppTranslationsForm
-        elif form_action == 'update':
-            return UpdateAppTranslationsForm
+        if form_action == 'create_update':
+            return CreateUpdateAppTranslationsForm
         elif form_action == 'push':
             return PushAppTranslationsForm
         elif form_action == 'pull':
             return PullAppTranslationsForm
-        elif form_action == 'backup':
-            return BackUpAppTranslationsForm
         elif form_action == 'delete':
             return DeleteAppTranslationsForm
 
 
-class CreateAppTranslationsForm(AppTranslationsForm):
-    form_action = 'create'
+class CreateUpdateAppTranslationsForm(AppTranslationsForm):
+    form_action = 'create_update'
+    update_existing_resource = forms.MultipleChoiceField(
+        choices=[
+            ('yes', 'Update existing resources'),
+        ],
+        widget=forms.CheckboxSelectMultiple(),
+        required=False,
+        initial='no',
+        help_text=gettext_lazy("Check this if you want to update an existing resource, instead of "
+                               "creating new ones. If you submit this form without this check, "
+                               "duplicate resources may be created.")
+    )
     source_lang = forms.ChoiceField(label=gettext_lazy("Source Language on Transifex"),
                                     choices=langcodes.get_all_langs_for_select(),
                                     initial="en"
                                     )
 
     def form_fields(self):
-        form_fields = super(CreateAppTranslationsForm, self).form_fields()
+        form_fields = super(CreateUpdateAppTranslationsForm, self).form_fields()
+        if self.form_action == 'create_update':
+            form_fields.append('update_existing_resource')
         form_fields.append(crispy.Field('source_lang', css_class="hqwebapp-select2"))
         return form_fields
 
 
-class UpdateAppTranslationsForm(CreateAppTranslationsForm):
-    form_action = 'update'
-
-
 class PushAppTranslationsForm(AppTranslationsForm):
     form_action = 'push'
 
     def form_fields(self):
         form_fields = super(PushAppTranslationsForm, self).form_fields()
+        # for some reason this works only works with both of these lines, not one or the other
```

Comment:
very much a nit: extra 'works' in this comment

Reply:
this line was taken out in a later commit btw